src/main/java/io/snyk/jenkins/tools/SnykInstallation.java

```diff
@@ -56,48 +53,55 @@ public String getReportExecutable(@Nonnull Launcher launcher) throws IOException
     return resolveExecutable(launcher, "snyk-to-html");
   }
 
-  private String resolveExecutable(Launcher launcher, String name)
-  throws IOException, InterruptedException {
+  private String resolveExecutable(Launcher launcher, String name) throws IOException, InterruptedException {
+    String home = Optional.ofNullable(getHome())
+      .orElseThrow(() -> new RuntimeException("Failed to find Snyk Executable. Installation Home is not configured."));
+
     return Optional.ofNullable(launcher.getChannel())
-      .orElseThrow(() -> new IOException("Failed to get snyk executable. Node does not support channels."))
-      .call(new MasterToSlaveCallable<String, IOException>() {
-        @Override
-        public String call() throws IOException {
-          Platform platform = Platform.current();
-          String filename = "snyk".equals(name)
-            ? platform.snykWrapperFileName
-            : platform.snykToHtmlWrapperFileName;
-
-          String root = Optional.ofNullable(getHome())
-            .orElseThrow(() -> new IOException(format(
-              "Failed to resolve executable <%s>. Could not find home.",
-              filename
-            )));
-
-          final Path executable = Paths.get(root).resolve(filename).toAbsolutePath();
-          if (!executable.toFile().exists()) {
-            throw new IOException(format("Could not find executable <%s>", filename));
-          }
-          return executable.toString();
-        }
-      });
+      .orElseThrow(() -> new RuntimeException("Failed to find Snyk Executable. Build Node does not support channels."))
+      .call(new ResolveExecutable(home, name));
```

Comment:
Moved this to its own named class to avoid serialisation warnings on build agents. They don't like anonymous classes.